opal/mca/btl/base/btl_base_am_rdma.c

```diff
@@ -567,21 +589,22 @@ static void mca_btl_base_rdma_queue_operation(mca_btl_base_module_t *btl,
 
     operation->is_queued = true;
     operation->atomic_response = atomic_response;
-    OPAL_THREAD_SCOPED_LOCK(&default_module.mutex,
-                            opal_list_append(&default_module.queued_responses, &operation->super));
+    OPAL_THREAD_SCOPED_LOCK(&default_component.mutex,
```

Comment:
What's the rational to use default_component here rather than the module?  Is this because not all btls provide the correct semantics for this thread-scoped mutex?

Reply:
It's about the data structure we're protection with the lock.  I decided to leave the `queued_responses` shared across BTLs by leaving it at the component level.  The disadvantage is that the queue draining takes a pretty big lock.  The advantage is that the progress function doesn't have to poll all am_rdma modules for work to do.  Given the upcoming changes in how the osc rdma component works, I expect there to be more BTLs with no work to do than BTLs with work to do at any given point, so a global structure seemed more efficient.